tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Http/WebRequest/HttpWebRequest_EndGetResponse_Integration.cs

```diff
@@ -94,9 +94,22 @@ public class HttpWebRequest_EndGetResponse_Integration
                         if (returnValue is HttpWebResponse response)
                         {
                             scope.Span.SetHttpStatusCode((int)response.StatusCode, isServer: false, Tracer.Instance.Settings);
+                            scope.Dispose();
                         }
+                        else if (exception is WebException { Status: WebExceptionStatus.ProtocolError, Response: HttpWebResponse exceptionResponse })
+                        {
+                            scope.Span.SetHttpStatusCode((int)exceptionResponse.StatusCode, isServer: false, Tracer.Instance.Settings);
 
-                        scope.DisposeWithException(exception);
+                            // SetHttpStatusCode will mark the span with an error if the StatusCode is within the configured range
+                            // Add the remaining error attributes without setting the Error property
+                            scope.Span.SetExceptionTags(exception);
+                            state.Scope.Dispose();
```

Comment:
I think this is maybe copy pasta?
```suggestion
```

Reply:
Yep good find, fixed in c909a6a0939f533badc2944484250e63b64b9311